python/spec/fixtures/pyproject_files/extra_source_explicit.toml

```diff
@@ -0,0 +1,17 @@
+[tool.poetry]
+name = "PythonProjects"
+version = "2.0.0"
+homepage = "https://github.com/roghu/py3_projects"
+license = "MIT"
+readme = "README.md"
+authors = ["Dependabot <support@dependabot.com>"]
+description = "Various small python projects."
+
+[tool.poetry.dependencies]
+python = "^3.7"
+requests = "2.18.0"
+
+[[tool.poetry.source]]
+name = "custom"
+url = "https://some.internal.registry.com/pypi/"
+priority = "explicit"
```

Comment:
Can we add a EOF character to this fixture?

Also, small question, my understanding is that this is not a common "realworld" file because the explicit source is not used by any dependency, but it's enough for our tests because the source information is already included in the `Dependency` instance that you are using for testing?

Reply:
> Can we add a EOF character to this fixture?
> 

Sure, will added

> Also, small question, my understanding is that this is not a common "realworld" file because the explicit source is not used by any dependency, but it's enough for our tests because the source information is already included in the `Dependency` instance that you are using for testing?

Indeed, this is not a real-world scenario. The test case utilizes the Dependency instance like other tests in index_finder_spec.rb. 

`extra_source_explicit.toml` has been intentionally crafted as a minimal example to replicate the bug #7918, and it is being tested in the unit test named **when set in a pyproject.toml file and marked as explicit.**

It might cause some confusion to reuse the TOML, so I will add a distinct TOML configuration in the unit test **when set in a pyproject.toml file and marked as explicit and specify with source**